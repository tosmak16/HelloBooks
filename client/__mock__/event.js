export const e = {
  target: {
    name: 'Tosmak',
    value: '10'
  },
};

export class event {
  static preventDefault() {

  }
}

