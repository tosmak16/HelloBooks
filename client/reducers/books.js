import { GET_ALL_BOOKS_FAILURE, GET_ALL_BOOKS_SUCCESS, GET_ALL_BOOKS_REQUEST } from '../actions/getBooks';

const initialState = {
  isFetching: false,
  isFetched: false,
  data: [],
  error: '',
};
export default (state = initialState, action = {}) => {
  switch (action.type) {
    case GET_ALL_BOOKS_REQUEST:
      return {
        isFetching: true,
        isFetched: false,
        data: [],
        error: '',
      };
    case GET_ALL_BOOKS_SUCCESS:
      return {
        isFetching: false,
        isFetched: true,
        data: action.data,
        error: '',
      };
    case GET_ALL_BOOKS_FAILURE:
      return {
        isFetching: false,
        isFetched: false,
        data: [],
        error: action.error,
      };
    default:
      return state;
  }
};
