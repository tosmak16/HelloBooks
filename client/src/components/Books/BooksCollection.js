import React from 'react';
// import { CardTitle, Card } from 'react-materialize';
import PropTypes from 'prop-types';

class BooksCollection extends React.Component {
  render() {
    const { data } = this.props;

    const litt = data.map(item => (<li key={ item.id } className="collection-item avatar">
      <img key={ item.id } src={ require(`../../../public/img/${item.image}`) } alt="" className="circle" />
      <span className="title">{item.bookTitle}</span>
      <p>{item.author}</p>
      <a href="#!" className="secondary-content"><i key={ item.id } style={{ color: 'orange' }} className="material-icons ">grade</i></a>
    </li>));

    return (
      <div className="col m10 offset-m2 col l10 offset-l2 col s12 main">
        <h4 className="page-header">{this.props.heading}</h4>
        <hr />
        <div className=" row ">
          <div className="">
            <ul className="collection ">
              {litt}
            </ul>
          </div>
        </div>
      </div>
    );
  }
}

BooksCollection.propTypes = {


};

export default BooksCollection;
