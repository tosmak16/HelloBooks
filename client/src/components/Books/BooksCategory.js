import React from 'react';
import PropTypes from 'prop-types';
import lodash from 'lodash';

import BooksCollection from './BooksCollection';


let sortedData = '';
let display = null;
class BooksCategory extends React.Component {
  componentWillMount() {
    if (this.props.data[0]) { sortedData = this.props.data[0]; }
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.data[0]) { sortedData = nextProps.data[0]; }
  }
  render() {
    if (sortedData.categoryData) {
      display = <BooksCollection heading={ `${sortedData.category} Books` } data={ sortedData.categoryData } />;
    }
    return (
      <div className="row">
        {!lodash.isEmpty(sortedData.categoryData) ? display : <h4>{''}</h4>}

      </div>
    );
  }
}
BooksCategory.propTypes = {
  data: PropTypes.array.isRequired,
};


export default BooksCategory;
