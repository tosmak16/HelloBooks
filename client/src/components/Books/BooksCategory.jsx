import React from 'react';
import PropTypes from 'prop-types';
import lodash from 'lodash';

import BooksCollection from './BooksCollection';

let sortedData = '';
let booksCollection = null;

/**
 * @class BooksCategory
 * @extends {React.Component}
 */
class BooksCategory extends React.Component {
  /**
   * @function componentWillMount
   * @memberof BooksCategory
   * @returns {void}
   */
  componentWillMount() {
    if (this.props.categoryData[0]) { sortedData = this.props.categoryData[0]; }
  }

  /**
   *
   * @function componentWillReceiveProps
   * @param {object} nextProps
   * @memberof BooksCategory
   * @returns {void}
   */
  componentWillReceiveProps(nextProps) {
    if (nextProps.categoryData[0]) { sortedData = nextProps.categoryData[0]; }
  }
  /**
   * @function render
   * @returns {view} containing book data selected by category
   * @memberof BooksCategory
   */
  render() {
    if (sortedData.categoryData) {
      booksCollection = <BooksCollection checkBookDetails={this.props.checkBookDetails} heading={`${sortedData.category} Books`} bookData={sortedData.categoryData} />;
    }
    return (
      <div className="row">
        {!lodash.isEmpty(sortedData.categoryData) ? booksCollection : <h4>{''}</h4>}
      </div>
    );
  }
}

BooksCategory.propTypes = {
  checkBookDetails: PropTypes.func.isRequired,
  categoryData: PropTypes.arrayOf(PropTypes.any).isRequired,
};


export default BooksCategory;
