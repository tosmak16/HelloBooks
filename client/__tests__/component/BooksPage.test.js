import React from 'react';
import { mount, shallow } from 'enzyme';
import render from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import expect from 'expect';
import thunk from 'redux-thunk';


import ConnectedBooksPage, { BooksPage } from '../../src/components/Books/BooksPage';
import localStorageMock from '../../__mock__/localStorage';
import books from '../../__mock__/book';
import BooksFilter from '../../src/components/Books/BooksFilter';
import BooksCollection from '../../src/components/Books/BooksCollection';
import BooksCategory from '../../src/components/Books/BooksCategory';
import BooksPhoto from '../../src/components/Books/BooksPhoto';
import SideBar from '../../src/components/SideBar';
import { e, event } from '../../__mock__/event';
import Cardbox from '../../src/components/Books/Cardbox';


const middlewares = [thunk]; // add your middlewares like `redux-thunk

const mockFuction = jest.fn();


describe('Test BooksPage component', () => {
  const initialState = {
    books: [{
      isFetching: false,
      isFetched: false,
      data: [],
      error: '',
    }],
    category: [{
      category: '',
      categoryData: '',
      selectedCategory: false,
    }],
    getFilteredBooks: [{
      filteredData: [],
    }]
  };
  const mockStore = configureMockStore(middlewares);
  let store;
  let wrapper;

  beforeEach(() => {
    store = mockStore(initialState);
    window.localStorage = localStorageMock;


    wrapper = mount(<Provider store={ store }><ConnectedBooksPage /></Provider>);
  });

  it('should test and take snapshot of booksPage', () => {
    const tree = render.create(<BooksPage
      categoryData={ [] }
      isFetched={ !initialState.books[0].isFetched }
      getbooks={ mockFuction }
      data={ [{ id: 1, name: 'hello' }] }
      showbooksByCategory={ mockFuction }
      filteredData={ [{ id: 1, bookTitle: 'love' }] }
      checkBookDetails={ mockFuction }
      searchbooks={ mockFuction }
    />);
    expect(tree).toMatchSnapshot();
  });

  it('should test and take snapshot of booksFilterPage', () => {
    const tree = render.create(<BooksFilter
      data={ [{ name: 'hello' }] }
      filteredData={ [{ id: 1, bookTitle: 'love' }] }
      checkBookDetails={ mockFuction }
      searchbooks={ mockFuction }
    />);
    expect(tree).toMatchSnapshot();
  });

  it('should test and take snapshot of booksCollectionPage', () => {
    const tree = render.create(<BooksCollection
      data={ [{ name: 'hello', image: '', id: 1 }] }
      checkBookDetails={ mockFuction }
      heading={ 'hello' }
    />);
    expect(tree).toMatchSnapshot();
  });


  it('should test and take snapshot of booksCategoryPage', () => {
    const tree = render.create(<BooksCategory
      categoryData={ [] }
      data={ [{ name: 'hello', image: '', id: 1 }] }
      checkBookDetails={ mockFuction }
    />);
    expect(tree).toMatchSnapshot();
  });

  it('should test and take snapshot of booksphotoPage', () => {
    const tree = render.create(<BooksPhoto
      data={ [{ name: 'hello', image: '', id: 1 }] }
      checkBookDetails={ mockFuction }
    />);
    expect(tree).toMatchSnapshot();
  });

  it('should test and take snapshot of Side Bar', () => {
    const tree = render.create(<SideBar
      data={ [{ name: 'hello', image: '', id: 1 }] }
      showbooksByCategory={ mockFuction }
    />);
    expect(tree).toMatchSnapshot();
  });

  it('should render the connected component', () => {
    expect(wrapper.find(ConnectedBooksPage).length).toEqual(1);
  });

  it(' checks Prop matches with initialState', () => {
    expect(wrapper.find(BooksPage).prop('data')).toEqual(initialState.books[0].data);
    expect(wrapper.find(BooksPage).prop('isFetched')).toEqual(initialState.books[0].isFetched);
    expect(wrapper.find(BooksPage).prop('categoryData')).toEqual(initialState.category);
    expect(wrapper.find(BooksPage).prop('filteredData')).toEqual(initialState.getFilteredBooks[0].filteredData);
  });

  it('should test for handle change function', () => {
    wrapper = shallow(<BooksFilter
      data={ [{ name: 'hello' }] }
      filteredData={ [{ id: 1, bookTitle: 'love' }] }
      checkBookDetails={ mockFuction }
      searchbooks={ mockFuction }
    />);
    wrapper.instance().handleFuction = mockFuction;
    wrapper.update();
    wrapper.instance().setState({
      filterBy: 'author',
      searchText: 'teh latehj'
    });
    wrapper.update();
    wrapper.instance().handleChange(e);
    wrapper.update();
  });

  it('should test for handle selected function', () => {
    wrapper = shallow(<BooksFilter
      data={ [{ name: 'hello' }] }
      filteredData={ [{ id: 1, bookTitle: 'love' }] }
      checkBookDetails={ mockFuction }
      searchbooks={ mockFuction }
    />);
    wrapper.instance().handleFuction = mockFuction;
    wrapper.update();
    wrapper.instance().handleSelected(e);
    wrapper.update();
    wrapper.instance().setState({ filterBy: 'title' });
    wrapper.update();
  });

  it('should test for functios in book filter', () => {
    wrapper = shallow(<BooksFilter
      data={ [{ name: 'hello' }] }
      filteredData={ [{ id: 1, bookTitle: 'love' }] }
      checkBookDetails={ mockFuction }
      searchbooks={ mockFuction }
    />);
    wrapper.instance().handleFuction = mockFuction;
    wrapper.update();
    wrapper.instance().handleSelected(e);
    wrapper.update();
    wrapper.instance().setState({
      ilterBy: '',
      searchText: '',
      error: '',
      pointer: false
    });
  });

  it('should test and take snapshot of Card box', () => {
    const component = render.create(<Cardbox
      item={{ name: 'hello', image: 'hello', id: 1 }}
      handleClick={ mockFuction }
      checkBookDetails={ mockFuction }
    />);

    expect(component).toMatchSnapshot();
  });

  it('should test for handle click function in sidbar component', () => {
    wrapper = shallow(<SideBar
      data={ [{ name: 'hello', image: '', id: 1 }] }
      showbooksByCategory={ mockFuction }
    />);
    wrapper.instance().handleFuction = mockFuction;
    wrapper.update();
    wrapper.instance().handleClick(e, false);
    wrapper.update();
    wrapper.instance().handleCollapse(e, false);
    wrapper.update();
  });

  it('should test for handle collapse function in sidbar component', () => {
    wrapper = shallow(<SideBar
      data={ [{ name: 'hello', image: '', id: 1 }] }
      showbooksByCategory={ mockFuction }
    />);
    wrapper.instance().handleFuction = mockFuction;
    wrapper.update();
    wrapper.instance().handleCollapse(e, false);
    wrapper.update();
  });

  it('should test for handle click function in cardbox component', () => {
    wrapper = shallow(<Cardbox
      item={{ name: 'hello', image: 'hello', id: 1 }}
      handleClick={ mockFuction }
      checkBookDetails={ mockFuction }
    />);
    wrapper.instance().handleFuction = mockFuction;
    wrapper.update();
    wrapper.instance().handleClick(e, false);
    wrapper.update();
  });

  it('should test for handle click function in BooksCollection component', () => {
    wrapper = shallow(<BooksCollection
      data={ [{ name: 'hello', image: 'l8.jpg', id: 1 }] }
      checkBookDetails={ mockFuction }
      heading={ 'hello' }
    />);
    wrapper.instance().handleFuction = mockFuction;
    wrapper.update();
    wrapper.instance().handleClick(e, false);
    wrapper.update();
  });

  it('should test for componeneWillReceiveProps function in BooksCategory component', () => {
    const nextProps = {
      data: [{ category: books }]
    };

    wrapper = shallow(<BooksCategory
      categoryData={ books[0] }
      data={ [{ name: 'hello', image: '', id: 1 }] }
      checkBookDetails={ mockFuction }
    />);
    wrapper.instance().handleFuction = mockFuction;
    wrapper.update();
    wrapper.instance().componentWillReceiveProps(nextProps);
    wrapper.update();
  });

  it('should test for componeneWillmount function in BooksCategory component', () => {
    const nextProps = {
      data: [{ category: books }]
    };

    wrapper = shallow(<BooksCategory
      categoryData={ books[0] }
      data={ [{ name: 'hello', image: '', id: 1 }] }
      checkBookDetails={ mockFuction }
    />);
    wrapper.instance().handleFuction = mockFuction;
    wrapper.update();
    wrapper.instance().componentWillMount();
    wrapper.update();
  });
});
