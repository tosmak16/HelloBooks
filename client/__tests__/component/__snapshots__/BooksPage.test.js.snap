// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test BooksPage component should test and take snapshot of Card box 1`] = `
<div
  id="card_div"
>
  <div
    className="container"
    id="img_holder"
  >
    <img
      alt="Avatar"
      className="image"
      id="book_img"
      src="l1.jpg"
      style={
        Object {
          "height": "200px",
          "width": "150px",
        }
      }
    />
    <div
      className="middle"
    >
      <label
        className="text"
        htmlFor="book_img"
        style={
          Object {
            "fontSize": "12px",
            "fontStyle": "bold",
          }
        }
      >
         
        THE PEOPLE IN THE TREE
      </label>
      <button
        className="text"
        name={1}
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "transparent",
            "width": "120px",
          }
        }
        type="btn-sm"
      >
        Read
      </button>
    </div>
  </div>
</div>
`;

exports[`Test BooksPage component should test and take snapshot of Side Bar 1`] = `
<div
  className="row"
>
  <div
    className="col s12 col m2 col l2 sidebar "
    id="sidenava"
  >
    <ul
      className="collapsible"
      data-collapsible="accordion"
    >
      <li>
        <div
          className="collapsible-header"
        />
        <div
          className="collapsible-body"
        />
        <div
          className="collapsible-body"
        />
        <div
          className="collapsible-body"
        />
      </li>
      <li>
        <div
          className="collapsible-header"
        />
        <div
          className="collapsible-body"
        />
        <div
          className="collapsible-body"
        />
        <div
          className="collapsible-body"
        />
        <div
          className="collapsible-body"
        />
        <div
          className="collapsible-body"
        />
        <div
          className="collapsible-body"
        />
      </li>
      <li>
        <div
          className="collapsible-header"
        />
        <div
          className="collapsible-body"
        />
        <div
          className="collapsible-body"
        />
        <div
          className="collapsible-body"
        />
        <div
          className="collapsible-body"
        />
        <div
          className="collapsible-body"
        />
        <div
          className="collapsible-body"
        />
      </li>
      <li>
        <div
          className="collapsible-header"
        />
        <div
          className="collapsible-body"
        />
        <div
          className="collapsible-body"
        />
        <div
          className="collapsible-body"
        />
        <div
          className="collapsible-body"
        />
      </li>
      <li>
        <div
          className="collapsible-header"
        />
        <div
          className="collapsible-body"
        />
        <div
          className="collapsible-body"
        />
        <div
          className="collapsible-body"
        />
        <div
          className="collapsible-body"
        />
      </li>
      <li>
        <div
          className="collapsible-header"
        />
        <div
          className="collapsible-body"
        />
        <div
          className="collapsible-body"
        />
        <div
          className="collapsible-body"
        />
      </li>
    </ul>
  </div>
</div>
`;

exports[`Test BooksPage component should test and take snapshot of booksCategoryPage 1`] = `
<div
  className="row"
/>
`;

exports[`Test BooksPage component should test and take snapshot of booksCollectionPage 1`] = `
<div
  className="col m10 offset-m2 col l10 offset-l2 col s12 main"
>
  <h4
    className="page-header"
  >
    hello
  </h4>
  <hr />
  <div
    className=" row "
  >
    <div
      className=""
    >
      <ul
        className="collection "
      >
        <li
          className="collection-item avatar"
        >
          <img
            alt=""
            className="circle"
            name={1}
            src=""
          />
          <span
            className="title"
          />
          <p />
          <button
            className="btn-sm btn-warning shop"
            id="wishbtn"
            name={1}
            onClick={[Function]}
            type="button"
          >
            Check details
          </button>
          <a
            className="secondary-content"
            href="#!"
          >
            <i
              className="material-icons "
              style={
                Object {
                  "color": "orange",
                }
              }
            >
              grade
            </i>
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`Test BooksPage component should test and take snapshot of booksFilterPage when bookIsFound is false 1`] = `
<div
  className="container-fluid"
  id="booksearchbar"
>
  <div
    className="row"
  >
    <div
      className="col l3  offset-l2  col m3  offset-m2  col s12 "
    >
      <div>
        <select
          className="browser-default input-field"
          name="filterBy"
          onChange={[Function]}
          value=""
        >
          <option
            defaultValue="bookTitle"
          >
            Search by
          </option>
          <option
            value="bookTitle"
          >
            Title
          </option>
          <option
            value="author"
          >
            Author
          </option>
          <option
            value="category"
          >
            Category
          </option>
          <option
            value="isbn"
          >
            ISBN
          </option>
        </select>
      </div>
    </div>
    <div
      className="col l6 col m6 col s12 "
    >
      <div
        className="input-field"
      >
        <input
          id="search"
          name="searchText"
          onChange={[Function]}
          required={true}
          style={
            Object {
              "backgroundColor": "white",
              "textAlign": "center",
            }
          }
          type="search"
          value=""
        />
        <label
          className="label-icon"
          htmlFor="search"
        >
          <i
            className="material-icons"
          >
            search
          </i>
        </label>
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col m10 offset-m2 col l10 offset-l2 col s12 main"
    >
      <h4
        className="page-header"
      >
        No search results
      </h4>
      <hr />
      <div
        className=" row "
      >
        <div
          className=""
        >
          <ul
            className="collection "
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Test BooksPage component should test and take snapshot of booksFilterPage when bookIsFound is true 1`] = `
<div
  className="container-fluid"
  id="booksearchbar"
>
  <div
    className="row"
  >
    <div
      className="col l3  offset-l2  col m3  offset-m2  col s12 "
    >
      <div>
        <select
          className="browser-default input-field"
          name="filterBy"
          onChange={[Function]}
          value=""
        >
          <option
            defaultValue="bookTitle"
          >
            Search by
          </option>
          <option
            value="bookTitle"
          >
            Title
          </option>
          <option
            value="author"
          >
            Author
          </option>
          <option
            value="category"
          >
            Category
          </option>
          <option
            value="isbn"
          >
            ISBN
          </option>
        </select>
      </div>
    </div>
    <div
      className="col l6 col m6 col s12 "
    >
      <div
        className="input-field"
      >
        <input
          id="search"
          name="searchText"
          onChange={[Function]}
          required={true}
          style={
            Object {
              "backgroundColor": "white",
              "textAlign": "center",
            }
          }
          type="search"
          value=""
        />
        <label
          className="label-icon"
          htmlFor="search"
        >
          <i
            className="material-icons"
          >
            search
          </i>
        </label>
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col m10 offset-m2 col l10 offset-l2 col s12 main"
    >
      <h4
        className="page-header"
      >
        Search results
      </h4>
      <hr />
      <div
        className=" row "
      >
        <div
          className=""
        >
          <ul
            className="collection "
          >
            <li
              className="collection-item avatar"
            >
              <img
                alt=""
                className="circle"
                name={1}
                src=""
              />
              <span
                className="title"
              >
                love
              </span>
              <p />
              <button
                className="btn-sm btn-warning shop"
                id="wishbtn"
                name={1}
                onClick={[Function]}
                type="button"
              >
                Check details
              </button>
              <a
                className="secondary-content"
                href="#!"
              >
                <i
                  className="material-icons "
                  style={
                    Object {
                      "color": "orange",
                    }
                  }
                >
                  grade
                </i>
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Test BooksPage component should test and take snapshot of booksPage 1`] = `
<div
  className=""
>
  <div
    className="row"
  >
    <div
      className=""
    >
      <div
        className="row"
      >
        <div
          className="col s12 col m2 col l2 sidebar "
          id="sidenava"
        >
          <ul
            className="collapsible"
            data-collapsible="accordion"
          >
            <li>
              <div
                className="collapsible-header"
              />
              <div
                className="collapsible-body"
              />
              <div
                className="collapsible-body"
              />
              <div
                className="collapsible-body"
              />
            </li>
            <li>
              <div
                className="collapsible-header"
              />
              <div
                className="collapsible-body"
              />
              <div
                className="collapsible-body"
              />
              <div
                className="collapsible-body"
              />
              <div
                className="collapsible-body"
              />
              <div
                className="collapsible-body"
              />
              <div
                className="collapsible-body"
              />
            </li>
            <li>
              <div
                className="collapsible-header"
              />
              <div
                className="collapsible-body"
              />
              <div
                className="collapsible-body"
              />
              <div
                className="collapsible-body"
              />
              <div
                className="collapsible-body"
              />
              <div
                className="collapsible-body"
              />
              <div
                className="collapsible-body"
              />
            </li>
            <li>
              <div
                className="collapsible-header"
              />
              <div
                className="collapsible-body"
              />
              <div
                className="collapsible-body"
              />
              <div
                className="collapsible-body"
              />
              <div
                className="collapsible-body"
              />
            </li>
            <li>
              <div
                className="collapsible-header"
              />
              <div
                className="collapsible-body"
              />
              <div
                className="collapsible-body"
              />
              <div
                className="collapsible-body"
              />
              <div
                className="collapsible-body"
              />
            </li>
            <li>
              <div
                className="collapsible-header"
              />
              <div
                className="collapsible-body"
              />
              <div
                className="collapsible-body"
              />
              <div
                className="collapsible-body"
              />
            </li>
          </ul>
        </div>
      </div>
    </div>
    <div
      className=""
      id="book_body"
    >
      <div
        className="row"
      >
        <div
          className="container-fluid"
          id="booksearchbar"
        >
          <div
            className="row"
          >
            <div
              className="col l3  offset-l2  col m3  offset-m2  col s12 "
            >
              <div>
                <select
                  className="browser-default input-field"
                  name="filterBy"
                  onChange={[Function]}
                  value=""
                >
                  <option
                    defaultValue="bookTitle"
                  >
                    Search by
                  </option>
                  <option
                    value="bookTitle"
                  >
                    Title
                  </option>
                  <option
                    value="author"
                  >
                    Author
                  </option>
                  <option
                    value="category"
                  >
                    Category
                  </option>
                  <option
                    value="isbn"
                  >
                    ISBN
                  </option>
                </select>
              </div>
            </div>
            <div
              className="col l6 col m6 col s12 "
            >
              <div
                className="input-field"
              >
                <input
                  id="search"
                  name="searchText"
                  onChange={[Function]}
                  required={true}
                  style={
                    Object {
                      "backgroundColor": "white",
                      "textAlign": "center",
                    }
                  }
                  type="search"
                  value=""
                />
                <label
                  className="label-icon"
                  htmlFor="search"
                >
                  <i
                    className="material-icons"
                  >
                    search
                  </i>
                </label>
              </div>
            </div>
          </div>
          <div
            className="row"
          />
        </div>
        <div
          className="row"
        />
        <div
          className="col m10 offset-m2 col l10 offset-l2 col s12 main"
        >
          <h4
            className="page-header"
          >
            Latest Books
          </h4>
          <hr />
          <div
            className="row"
          >
            <div
              className="col m5 col l3 col s6"
            >
              <div
                id="card_div"
              >
                <div
                  className="container"
                  id="img_holder"
                >
                  <img
                    alt="Avatar"
                    className="image"
                    id="book_img"
                    src={undefined}
                    style={
                      Object {
                        "height": "200px",
                        "width": "150px",
                      }
                    }
                  />
                  <div
                    className="middle"
                  >
                    <label
                      className="text"
                      htmlFor="book_img"
                      style={
                        Object {
                          "fontSize": "12px",
                          "fontStyle": "bold",
                        }
                      }
                    >
                       
                    </label>
                    <button
                      className="text"
                      name={1}
                      onClick={[Function]}
                      style={
                        Object {
                          "backgroundColor": "transparent",
                          "width": "120px",
                        }
                      }
                      type="btn-sm"
                    >
                      Read
                    </button>
                  </div>
                </div>
              </div>
               
            </div>
          </div>
        </div>
        <div
          className="col m10 offset-m2 col l10 offset-l2 col s12 main"
        >
          <h4
            className="page-header"
          >
            Available books
          </h4>
          <hr />
          <div
            className=" row "
          >
            <div
              className=""
            >
              <ul
                className="collection "
              >
                <li
                  className="collection-item avatar"
                >
                  <img
                    alt=""
                    className="circle"
                    name={1}
                    src=""
                  />
                  <span
                    className="title"
                  />
                  <p />
                  <button
                    className="btn-sm btn-warning shop"
                    id="wishbtn"
                    name={1}
                    onClick={[Function]}
                    type="button"
                  >
                    Check details
                  </button>
                  <a
                    className="secondary-content"
                    href="#!"
                  >
                    <i
                      className="material-icons "
                      style={
                        Object {
                          "color": "orange",
                        }
                      }
                    >
                      grade
                    </i>
                  </a>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Test BooksPage component should test and take snapshot of booksphotoPage 1`] = `
<div
  className="col m10 offset-m2 col l10 offset-l2 col s12 main"
>
  <h4
    className="page-header"
  >
    Latest Books
  </h4>
  <hr />
  <div
    className="row"
  >
    <div
      className="col m5 col l3 col s6"
    >
      <div
        id="card_div"
      >
        <div
          className="container"
          id="img_holder"
        >
          <img
            alt="Avatar"
            className="image"
            id="book_img"
            src="l1.jpg"
            style={
              Object {
                "height": "200px",
                "width": "150px",
              }
            }
          />
          <div
            className="middle"
          >
            <label
              className="text"
              htmlFor="book_img"
              style={
                Object {
                  "fontSize": "12px",
                  "fontStyle": "bold",
                }
              }
            >
               
              THE PEOPLE IN THE TREE
            </label>
            <button
              className="text"
              name={1}
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "width": "120px",
                }
              }
              type="btn-sm"
            >
              Read
            </button>
          </div>
        </div>
      </div>
       
    </div>
    <div
      className="col m5 col l3 col s6"
    >
      <div
        id="card_div"
      >
        <div
          className="container"
          id="img_holder"
        >
          <img
            alt="Avatar"
            className="image"
            id="book_img"
            src="l2.jpg"
            style={
              Object {
                "height": "200px",
                "width": "150px",
              }
            }
          />
          <div
            className="middle"
          >
            <label
              className="text"
              htmlFor="book_img"
              style={
                Object {
                  "fontSize": "12px",
                  "fontStyle": "bold",
                }
              }
            >
               
              THE LATE PARADE
            </label>
            <button
              className="text"
              name={2}
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "width": "120px",
                }
              }
              type="btn-sm"
            >
              Read
            </button>
          </div>
        </div>
      </div>
       
    </div>
    <div
      className="col m5 col l3 col s6"
    >
      <div
        id="card_div"
      >
        <div
          className="container"
          id="img_holder"
        >
          <img
            alt="Avatar"
            className="image"
            id="book_img"
            src="l3.jpg"
            style={
              Object {
                "height": "200px",
                "width": "150px",
              }
            }
          />
          <div
            className="middle"
          >
            <label
              className="text"
              htmlFor="book_img"
              style={
                Object {
                  "fontSize": "12px",
                  "fontStyle": "bold",
                }
              }
            >
               
              I AM THE WOLF
            </label>
            <button
              className="text"
              name={3}
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "width": "120px",
                }
              }
              type="btn-sm"
            >
              Read
            </button>
          </div>
        </div>
      </div>
       
    </div>
    <div
      className="col m5 col l3 col s6"
    >
      <div
        id="card_div"
      >
        <div
          className="container"
          id="img_holder"
        >
          <img
            alt="Avatar"
            className="image"
            id="book_img"
            src="l4.jpg"
            style={
              Object {
                "height": "200px",
                "width": "150px",
              }
            }
          />
          <div
            className="middle"
          >
            <label
              className="text"
              htmlFor="book_img"
              style={
                Object {
                  "fontSize": "12px",
                  "fontStyle": "bold",
                }
              }
            >
               
              THE PALE NORTH
            </label>
            <button
              className="text"
              name={4}
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "width": "120px",
                }
              }
              type="btn-sm"
            >
              Read
            </button>
          </div>
        </div>
      </div>
       
    </div>
    <div
      className="col m5 col l3 col s6"
    >
      <div
        id="card_div"
      >
        <div
          className="container"
          id="img_holder"
        >
          <img
            alt="Avatar"
            className="image"
            id="book_img"
            src="l5.jpg"
            style={
              Object {
                "height": "200px",
                "width": "150px",
              }
            }
          />
          <div
            className="middle"
          >
            <label
              className="text"
              htmlFor="book_img"
              style={
                Object {
                  "fontSize": "12px",
                  "fontStyle": "bold",
                }
              }
            >
               
              ADULTOLESENCE
            </label>
            <button
              className="text"
              name={5}
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "width": "120px",
                }
              }
              type="btn-sm"
            >
              Read
            </button>
          </div>
        </div>
      </div>
       
    </div>
    <div
      className="col m5 col l3 col s6"
    >
      <div
        id="card_div"
      >
        <div
          className="container"
          id="img_holder"
        >
          <img
            alt="Avatar"
            className="image"
            id="book_img"
            src="l6.jpg"
            style={
              Object {
                "height": "200px",
                "width": "150px",
              }
            }
          />
          <div
            className="middle"
          >
            <label
              className="text"
              htmlFor="book_img"
              style={
                Object {
                  "fontSize": "12px",
                  "fontStyle": "bold",
                }
              }
            >
               
              WHAT IT MEANS WHEN A MAN FALL FROM SKY
            </label>
            <button
              className="text"
              name={6}
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "width": "120px",
                }
              }
              type="btn-sm"
            >
              Read
            </button>
          </div>
        </div>
      </div>
       
    </div>
    <div
      className="col m5 col l3 col s6"
    >
      <div
        id="card_div"
      >
        <div
          className="container"
          id="img_holder"
        >
          <img
            alt="Avatar"
            className="image"
            id="book_img"
            src="l7.jpg"
            style={
              Object {
                "height": "200px",
                "width": "150px",
              }
            }
          />
          <div
            className="middle"
          >
            <label
              className="text"
              htmlFor="book_img"
              style={
                Object {
                  "fontSize": "12px",
                  "fontStyle": "bold",
                }
              }
            >
               
              THE SONG AND THE SILENCE
            </label>
            <button
              className="text"
              name={7}
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "width": "120px",
                }
              }
              type="btn-sm"
            >
              Read
            </button>
          </div>
        </div>
      </div>
       
    </div>
    <div
      className="col m5 col l3 col s6"
    >
      <div
        id="card_div"
      >
        <div
          className="container"
          id="img_holder"
        >
          <img
            alt="Avatar"
            className="image"
            id="book_img"
            src="l8.jpg"
            style={
              Object {
                "height": "200px",
                "width": "150px",
              }
            }
          />
          <div
            className="middle"
          >
            <label
              className="text"
              htmlFor="book_img"
              style={
                Object {
                  "fontSize": "12px",
                  "fontStyle": "bold",
                }
              }
            >
               
              THE ILIAD HOMMER
            </label>
            <button
              className="text"
              name={8}
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "width": "120px",
                }
              }
              type="btn-sm"
            >
              Read
            </button>
          </div>
        </div>
      </div>
       
    </div>
  </div>
</div>
`;
